### ISSUE TITLE
PG - expandable object returns only base class properties.

### ISSUE THREAD

**Issue by xceedsoftware** [xceedsoftware/wpftoolkit#199]
<b>masyura[CodePlex]</b> <br />Let me explain the problem using some code snippets.
nbsp

I have interface:
nbsp
public interface IIndicator
{
string Name { get; set; }
}
nbsp I have the class which implements that interface:
nbsp
public class LengthIndicator : IIndicator
{
public int Length { set; get;}
public string Name { set; get;}
}
nbsp I have an object I want to edit via PG
nbsp
public class SampleUnit
{
public SampleUnit()
{
    Indicator = new LengthIndicator() {Length = 20, Name = quot234quot};
}
nbsp
[ExpandableObject]
[DisplayName(quotИндикаторquot)]
[Description(quotПар&#1084;етры индикатора.quot)]
[Category(quotПара&#1084;етрыquot)]
public IIndicator Indicator { get; set; }
}
nbsp
The problem is only properties from the IIndicator interface are displayed.

**Comment by xceedsoftware** [comment]
<b>masyura[CodePlex]</b> <br />Perfect. Thanks.

**Comment by xceedsoftware** [comment]
Resolved with changeset 81966.

**Comment by xceedsoftware** [comment]
<b>masyura[CodePlex]</b> <br />I see the TypeDescriptor.GetProperties returns the PropertyDescriptor for the Indicator as IIndicator instead of LengthIndicator even if I changed from descriptors = TypeDescriptor.GetProperties(instance.GetType()); to descriptors = TypeDescriptor.GetProperties(instance)
 in the PropertyGridUtilities.cs. 

I've found this article 
http://social.msdn.microsoft.com/Forums/en-US/netfxbcl/thread/2998956c-8bcc-4ebb-b926-7b1c1ba24d8f/ useful to get this problem solved, but didn't get luck with that.


Thanks for the help.
